h2/src/main/org/h2/mvstore/tx/TransactionMap.java

```diff
@@ -355,16 +372,19 @@ private V set(Object key, TxDecisionMaker<K,V> decisionMaker) {
                 return res;
             }
             decisionMaker.reset();
+            if (timeoutMillis < -1) {
```

Comment:
rather use timeoutMillis == -2, since that is the only magic value being tested for